src/redux/books/bookSlice.js

```diff
@@ -1,39 +1,64 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/Vc7rXRyGClD25RCBUtWt/books';
 
 const initialState = {
-  booksList: [{
-    item_id: 'item1',
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-    category: 'Fiction',
-  },
-  {
-    item_id: 'item2',
-    title: 'Anna Karenina',
-    author: 'Leo Tolstoy',
-    category: 'Fiction',
-  },
-  {
-    item_id: 'item3',
-    title: 'The Selfish Gene',
-    author: 'Richard Dawkins',
-    category: 'Nonfiction',
-  }],
+  booksList: [],
+  status: 'idle',
+  error: null,
 };
 
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  try {
+    const response = await axios.get(URL);
+    return response.data;
+  } catch (error) {
+    return error.message;
+  }
+});
+
+export const addNewBook = createAsyncThunk('books/addBook', async (newBookAdded) => {
+  try {
+    const response = await axios.post(URL, newBookAdded);
+    return response.data;
+  } catch (error) {
+    return error.message;
+  }
+});
+
+export const deleteBook = createAsyncThunk('books/deleteBook', async (bookId) => {
+  try {
+    const response = await axios.delete(`${URL}/${bookId}`);
+    return response.data;
+  } catch (error) {
+    return error.message;
+  }
+});
+
 const bookSlice = createSlice({
   name: 'books',
   initialState,
-  reducers: {
-    addBook: (state, action) => {
-      const newBook = action.payload;
-      state.booksList.push(newBook);
-    },
-    removeBook: (state, action) => {
-      state.booksList = state.booksList.filter((book) => book.item_id !== action.payload);
-    },
+  reducers: {},
+  extraReducers(builder) {
+    builder
+      .addCase(fetchBooks.pending, (state) => {
+        state.status = 'loading';
+      })
+      .addCase(fetchBooks.fulfilled, (state, action) => {
+        state.status = 'succeeded';
+        state.booksList = action.payload;
+      })
+      .addCase(fetchBooks.rejected, (state, action) => {
+        state.status = 'failed';
+        state.error = action.error.message;
+      });
   },
 });
 
-export const { removeBook, addBook } = bookSlice.actions;
+export const selectAllBooks = (state) => state.book.booksList;
+export const getBooksStatus = (state) => state.book.status;
+export const getBooksError = (state) => state.book.error;
+
+// export const { removeBook, addBook } = bookSlice.actions;
```

Comment:
- [OPTIONAL] Please do not leave unused code as a comment in your codebase. this will make your code much noisier, It is always a good practice to keep your code clean from such comments. 😉 👍🏻 Since it's the only one you have I'm gonna approve this PR but I request you remove it.

Reply:
Thank you so much @BenMukebo!

I  got it.